apps/prairielearn/src/pages/publicQuestionPreview/publicQuestionPreview.ts

```diff
@@ -25,6 +26,16 @@ async function setLocals(req, res) {
   res.locals.authz_data = { user: res.locals.user };
   res.locals.course = await selectCourseById(req.params.course_id);
   res.locals.question = await selectQuestionById(req.params.question_id);
+
+  const disablePublicWorkspaces = await features.enabledFromLocals(
+    'disable-public-workspaces',
+    res.locals,
+  );
+
+  if (res.locals.question.workspace_image && disablePublicWorkspaces) {
+    throw new Error('Access denied');
```

Comment:
Let's use a semantic error instead of an implicit 500 so that this doesn't show up in our error logs/metrics. An example:

https://github.com/PrairieLearn/PrairieLearn/blob/1624de88649627da1f76b4ac32db5e0c0b79acbf/apps/prairielearn/src/lib/course.ts#L40

Reply:
sure, done!